modules/cnf-performing-end-to-end-tests-for-platform-verification.adoc

```diff
@@ -89,78 +89,298 @@ This allows your `kubeconfig` file to be consumed from inside the running contai
+Running Suite: CNF Features e2e integration tests
+=================================================
+Random Seed: 1628762033
+Will run 1 of 138 specs
+
+SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
+------------------------------
+• [SLOW TEST:26.144 seconds]
+[performance] Latency Test
+/go/src/github.com/openshift-kni/cnf-features-deploy/vendor/github.com/openshift-kni/performance-addon-operators/functests/4_latency/latency.go:84
+  with the hwlatdetect image
+  /go/src/github.com/openshift-kni/cnf-features-deploy/vendor/github.com/openshift-kni/performance-addon-operators/functests/4_latency/latency.go:224
+    should succeed
+    /go/src/github.com/openshift-kni/cnf-features-deploy/vendor/github.com/openshift-kni/performance-addon-operators/functests/4_latency/latency.go:232
+------------------------------
+SSSSSSSSSSSSSSSSSSSSSS
+
+detector: tracer
+   parameters:
+        Latency threshold: 1us <2>
+        Sample window:     10000000us
+        Sample width:      950000us
+     Non-sampling period:  9050000us
+        Output File:       None
+
+Starting test
+test finished
+Max Latency: 35us <3>
+Samples recorded: 2
+Samples exceeding threshold: 2
+ts: 1628174377.074638224, inner:20, outer:35
+ts: 1628174387.359881340, inner:21, outer:34
+; err: exit status 1
+goroutine 1 [running]:
+k8s.io/klog.stacks(0xc000070200, 0xc000106400, 0x21b, 0x3c2)
+	/remote-source/app/vendor/k8s.io/klog/klog.go:875 +0xb9
+k8s.io/klog.(*loggingT).output(0x5bed00, 0xc000000003, 0xc00010a0e0, 0x53ea81, 0x7, 0x33, 0x0)
+	/remote-source/app/vendor/k8s.io/klog/klog.go:826 +0x35f
+k8s.io/klog.(*loggingT).printf(0x5bed00, 0x3, 0x5082da, 0x33, 0xc000113f58, 0x2, 0x2)
+	/remote-source/app/vendor/k8s.io/klog/klog.go:707 +0x153
+k8s.io/klog.Fatalf(...)
+	/remote-source/app/vendor/k8s.io/klog/klog.go:1276
+main.main()
+	/remote-source/app/cnf-tests/pod-utils/hwlatdetect-runner/main.go:51 +0x897
+----
+<1> The docker arguments provided by the user. 
+//check docker run arguments?
+<2> The latency threshold configured by the user using the `MAX_LATENCY` or the `HWLATDETECT_MAX_LATENCY` environment variables.
+<3> The maximum latency value measured during the test.
+
+[id="cnf-performing-end-to-end-tests-running-cyclictest"]
+==== Running `cyclictest`
+
+To perform the `cyclictest`, run the following command:
+
+[source, terminal]
+----
+$ docker run -v $(pwd)/:/kubeconfig -e KUBECONFIG=/kubeconfig/kubeconfig -e LATENCY_TEST_RUN=true -e DICOVERY_MODE=true -e LATENCY_TEST_CPUS=10 -e LATENCY_TEST_RUNTIME=600 -e MAXIMUM_LATENCY=20  registry.redhat.io/openshift4/cnf-tests-rhel8:v4.9 /usr/bin/test-run.sh -ginkgo.focus="cyclictest"
+----
+
+The above command runs the `cyclictest` tool for 10 minutes (600 seconds). The test runs successfully when the maximum observed latency is lower than `MAXIMUM_LATENCY` (20 us) and the command line displays `SUCCESS!`.
```

Comment:
(20 us)? Is that supposed to be milliseconds (20 ms)? Or is it microseconds (20 μs)?

Reply:
microseconds